conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/SerDe.java

```diff
@@ -36,15 +36,15 @@
 /** Serialization functionality consumed by generated code. */
 public interface SerDe {
```

Comment:
Could we maybe extract the param deserializers into a ParamDeserializers interface and make SerDe extends it ?

Reply:
I don't think that buys us a lot, the logical extreme is to put everything back on the `ConjureRuntime` interface. I'd prefer to have as few separate interfaces in the lib module as possible, which will make it easier to think about backcompat.

Reply:
If we were to go on an extreme with this, you could have BodySerDe, ParamSerDe and StaticTypeSerDe and have SerDe extends all three.

Reply:
Yep, weighted against adding more interfaces to `lib` I don't think it's super helpful, but I don't have a strong opinion if you disagree.

Reply:
Personally I don't really mind more interface as long as they allow better compartmentalization.
Maybe it's just missing ,ore explicit javadoc (you could potentially have an handler in the javadoc that use all feature of SerDe) but it is not trivial to follow on a first read.
I feel like intuitively reading this you would think there is a lot of factorization that could be done with all those deserialize methods (even though it's not the case). So some comments about some of the design choice would help.

Reply:
I think we can punt this until later, the `Encoding` interface is public, but only to allow us to test support for other content types outside of this repository. It shouldn't be considered a major break for the runtime module to make changes to Encoding because implementors must understand the risk. I'll add that to the javadoc :-)

Reply:
I think you're right on this one. perhaps we should move the query/path/header deserializaton methods to a `Plain` interface since these are values which use the conjure Plain format. e.g. `runtime.plain().deserializeString(in)`, thoughts?

Reply:
alternatively add `Plain` to the method names, e.g. `runtime.serde().deserializePlainString(in)`
It feels odd not to put those on the `SerDe` interface

Reply:
I would prefer `runtime.serde().plain().deserializeString(in)` and make `Plain` a field of handlers.

Reply:
We can continue this discussion in #260 (target of this PR) since this PR doesn't modify SerDe.<plain deserializer functions>